ansible/idr-microservices.yml

```diff
@@ -0,0 +1,131 @@
+
+  # pre_tasks:
+  #   - debug: var=inventory_hostname
+  #   - debug: msg="{{ hostvars[inventory_hostname] | to_yaml }}"
+  #   - fail: msg=stopping
+
+  roles:
+
+    - role: ome.docker
+      docker_use_ipv4_nic_mtu: True
+
+  tasks:
+
+    - name: Install docker-python
+      become: true
+      yum:
+        name: docker-python
+        state: present
+
+    - name: Get IPs of hosts
+      set_fact:
+        omero_host_ip: >-
+          {{ hostvars[inventory_hostname]['ansible_' + (idr_net_iface |
+             default('eth0'))]['ipv4']['address']
+          }}
+        omero_db_host_ansible: >-
+          {{ hostvars[groups[idr_environment |
+             default('idr') + '-database-hosts'][0]][
+             'ansible_' + (idr_net_iface | default('eth0'))]['ipv4']['address']
+          }}
+
+    - name: omero-ms config directories
+      become: true
+      file:
+        path: /etc/omero-ms-{{ item }}
+        state: directory
+      with_items:
+        - thumbnail
+        - imageregion
+        - zarr
+
+    - name: omero-ms config files
+      become: true
+      template:
+        src: templates/omero-ms-{{ item }}-config.yml.j2
+        dest: /etc/omero-ms-{{ item }}/config.yml
+      with_items:
+        - thumbnail
+        - imageregion
+
+    - name: omero-ms custom entrypoints
+      become: true
+      template:
+        src: templates/omero-ms-{{ item }}-entrypoint.sh.j2
+        dest: /etc/omero-ms-{{ item }}/omero-ms-{{ item }}-entrypoint.sh
+        mode: u=rwx,g=rx,o=rx
+      with_items:
+        - imageregion
+        - zarr
+
+    - name: Run docker omero-ms-thumbnail
+      become: true
+      docker_container:
+        image: openmicroscopy/omero-ms-thumbnail:0.5.3
+        name: omero-ms-thumbnail
+        published_ports:
+          - "{{ omero_ms_thumbnail_port }}:8080"
+        state: started
+        restart_policy: always
+        volumes:
+          - "/etc/omero-ms-thumbnail/config.yml:/opt/ms/conf/config.yaml:ro"
+
+    - name: Run docker omero-ms-imageregion
+      become: true
+      docker_container:
+        image: manics/docker-example-omero-microservices-imageregion:0.5.2-0
+        name: omero-ms-imageregion
+        # Override the entrypoint to create a fixed UID and GID
+        entrypoint: /opt/ms/bin/omero-ms-imageregion-entrypoint.sh
+        published_ports:
+          - "{{ omero_ms_imageregion_port }}:8080"
+        state: started
+        restart_policy: always
+        volumes:
+          - "/etc/omero-ms-imageregion/config.yml:/opt/ms/conf/config.yaml:ro"
+          - "/etc/omero-ms-imageregion/omero-ms-imageregion-entrypoint.sh:/opt/ms/bin/omero-ms-imageregion-entrypoint.sh:ro"
+          - "/data:/data:ro"
+          - "/opt/omero/server/OMERO.server/lib/scripts:/opt/omero/server/OMERO.server/lib/scripts:ro"
+          - "/uod/idr:/uod/idr:ro"
+          - "/nfs/bioimage:/nfs/bioimage:ro"
+          - "/nfs/biostudies:/nfs/biostudies:ro"
+
+    - name: Run docker omero-ms-zarr
```

Comment:
What's the role of `omero-ms-zarr` here? It is required for the functionality of the other microservices? It provides zarr API endpoints which aren't available in the current IDR deployment.

Reply:
It's no longer needed. In fact, I'd assume we should strip this down to only the endpoints we need if we're going to test it.

Reply:
I think the only one we need is `imageregion`. I don't think we need thumbnails at this point (could be added later if desired).